### ISSUE TITLE
Multiple adapter per engine

### ISSUE THREAD

**Issue by brunowego** [ember-engines/ember-engines#350]
Hello, I need to connect each engine to a different adapter. At that moment after connecting to the first adapter and changing the engine the adapter does not change. It is possible?

**Comment by brunowego** [comment]
To solve this issue, I use the below code, written by @alexferreira. This is so strange approach.

```js
// app/adapters/application.js
import DS from 'ember-data';
import config from 'apollo/config/environment';
import DataAdapterMixin from 'ember-simple-auth/mixins/data-adapter-mixin';
import Ember from 'ember';
import computed from 'ember-computed';

export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
  init() {
    let router = Ember.getOwner(this)
      .application.__container__.lookup('router:main');

    if (!this.get('currentPath')) {
      this.set('currentPath', location.pathname.split('/')[1]);
    }

    Ember.addObserver(router, 'currentPath', () => {
      this.set('currentPath', router.get('currentPath'));
    });
  },
  host: computed('currentPath', function(){
    let pathSplit = this.get('currentPath').split('.');

    if (pathSplit[0] == 'enterprise') {
      return 'https://enterprise.secret.lvh.me';
    }

    return 'https://api.secret.lvh.me';
  }),
  authorizer: 'authorizer:token'
});
```

Perhaps this can help explain my necessity. For now, this works.

**Comment by trentmwillis** [comment]
> connect each engine to a different adapter

When you say "each engine" are you referring to different Engine classes? Or the same Engine class but mounted in more than one location in your application?

**Comment by alexferreira** [comment]
Hello Trent, he is referring to different Engine classes.

He has a model user in the engine a and b, and wants to make a findAll in the user in the engine he accesses the host 'api.secret.lvh.me/users/1' and when doing the same request in engine b he accesses the host 'enterprise.secret.lvh.me/users/1'.

The suggestion I gave above was due to the fact of creating an application adapter for different engine in the same application will end up overlapping, so I arrived at the solution above that is not elegant but that already solves until having a better solution.
